packages/web/src/common/store/cache/tracks/sagas.ts

```diff
@@ -90,6 +90,34 @@ function* watchAdd() {
   )
 }
 
+function* watchUpdate() {
+  yield* takeEvery(
+    cacheActions.UPDATE,
+    function* (action: ReturnType<typeof cacheActions.update>) {
+      const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+      const isStreamPrefetchEnabled = yield* call(
+        getFeatureEnabled,
+        FeatureFlags.PREFETCH_STREAM_URLS
+      )
+      if (!isStreamPrefetchEnabled) return
+
+      if (action.kind === Kind.TRACKS) {
```

Comment:
mind having this as the first line of the saga and early returning if not a track action?